contracts/debtAllocators/YieldManager/YieldManager.sol

```diff
@@ -0,0 +1,388 @@
+        address strategy;
+        // Debt for the strategy to end with.
+        // Can take 79 Billion 18 decimal tokens.
+        uint96 newDebt;
+    }
+
+    /// @notice Only allow the sender to be an allocator if not opened.
+    modifier onlyProposersOrOpen() {
+        _isProposerOrOpen();
+        _;
+    }
+
+    /// @notice Check if it has been opened or is an allocator.
+    function _isProposerOrOpen() internal view {
+        require(proposer[msg.sender] || open, "!allocator or open");
+    }
+
+    uint256 internal constant MAX_BPS = 10_000;
+
+    /// @notice Contract that holds the logic and oracles for each strategy.
+    AprOracle internal constant aprOracle =
+        AprOracle(0xF012fBb9283e03994A7829fCE994a105cC066c14);
+
+    /// @notice Flag to set to allow anyone to propose allocations.
+    bool public open;
+
+    /// @notice Address that should hold the strategies `management` role.
+    address public immutable keeper;
+
+    /// @notice Addresses that are allowed to propose allocations.
+    mapping(address => bool) public proposer;
+
+    /// @notice Mapping for vaults that can be allocated for => its debt allocator.
+    mapping(address => address) public vaultAllocator;
+
+    constructor(address _governance, address _keeper) Governance(_governance) {
+        keeper = _keeper;
+    }
+
+    /**
+     * @notice Update a `_vault`s allocation of debt.
+     * @dev This takes the address of a vault and an array of
+     * its strategies and their specific allocation.
+     *
+     * The `_newAllocations` array should:
+     *   - Be ordered so that all debt decreases are at the beginning of the array
+     *       and debt increases at the end.
+     *
+     * This will not do any APR checks and assumes the sender has completed
+     * any and all necessary checks before sending.
+     *
+     * @param _vault The address of the vault to propose an allocation for.
+     * @param _newAllocations Array of strategies and their new proposed allocation.
+     */
+    function updateAllocationPermissioned(
+        address _vault,
+        Allocation[] memory _newAllocations
+    ) public virtual onlyGovernance {
+        // Move funds
+        _allocate(_vault, _newAllocations);
+    }
+
+    /**
+     * @notice Update a `_vault`s allocation of debt.
+     * @dev This takes the address of a vault and an array of
+     * its strategies and their specific allocation.
+     *
+     * The `_newAllocations` array should:
+     *   - Contain all strategies that hold any amount of debt from the vault
+     *       even if the debt wont be adjusted in order to get the correct
+     *       on chain APR.
+     *   - Be ordered so that all debt decreases are at the beginning of the array
+     *       and debt increases at the end.
+     *
+     * It is expected that the proposer does all needed checks for values such
+     * as max_debt, maxWithdraw, min total Idle etc. that are enforced on debt
```

Comment:
"max_debt, maxWithdraw, min total Idle etc." this is like 3 different inconsistent formats

Reply:
updated it to make a little more clear https://github.com/yearn/vault-periphery/pull/30/commits/fc3b236c152d06d93bc4dcffff3b8bd14d6608b4

Still snake case for the vault stuff though